pennylane_qiskit/converter.py

```diff
@@ -0,0 +1,167 @@
+    been bound.
+
+    Args:
+        param: the parameter to be checked
+
+    Returns:
+        param: the parameter after the check
+    """
+    if isinstance(param, Parameter):
+        raise ValueError("The parameter {} was not bound correctly.".format(param))
+    return param
+
+
+def load(quantum_circuit: QuantumCircuit):
+    """Returns a PennyLane template created based on the input QuantumCircuit or QASM string.
+    Warnings are created for each of the QuantumCircuit instructions that were
+    not incorporated in the PennyLane template.
+
+    Args:
+        quantum_circuit (qiskit.QuantumCircuit): the QuantumCircuit to be converted
+
+    Returns:
+        function: the new PennyLane template
+    """
+
+    def _function(params: dict = None, wires: list = None):
+        """Returns a PennyLane template created based on the input QuantumCircuit.
+        Warnings are created for each of the QuantumCircuit instructions that were
+        not incorporated in the PennyLane template.
+
+        Args:
+            params (dict): specifies the parameters that need to bound in the
+            QuantumCircuit
+
+            wire_shift (list): the shift to be made with respect to the wires
+            already specifiec in the QuantumCircuit
+            e.g. if the QuantumCircuit acts on wires [0, 1] and if wire_shift == 1,
+            then the returned Pennylane template will act on wires [1, 2]
+
+        Returns:
+            function: the new PennyLane template
+        """
+
+        if not isinstance(quantum_circuit, QuantumCircuit):
+            raise ValueError("The circuit {} is not a valid Qiskit QuantumCircuit.".format(quantum_circuit))
+
+        qc = quantum_circuit.bind_parameters(params) if params is not None else quantum_circuit
+
+        # Wires from a qiskit circuit are unique w.r.t. a register name and a qubit index
+        qc_wires = [(q.register.name, q.index) for q in quantum_circuit.qubits]
+
+        if wires is None:
+            wire_map = dict(zip(qc_wires, range(len(qc_wires))))
+        elif len(qc_wires) == len(wires):
+            wire_map = dict(zip(qc_wires, wires))
+        else:
+            raise WiresError("The specified number of wires - {} - does not match "
+                             "the number of wires the loaded quantum circuit acts on.".format(len(wires)))
+
+        # Processing the dictionary of parameters passed
+        for op in qc.data:
+
+            instruction_name = op[0].__class__.__name__
+
+            operation_wires = [wire_map[(qubit.register.name, qubit.index)] for qubit in op[1]]
+
+            if instruction_name in inv_map and inv_map[instruction_name] in pennylane_ops.ops:
+
+                operation_name = inv_map[instruction_name]
+
+                operation = getattr(pennylane_ops, operation_name)
+
+                parameters = [check_parameter_bound(param) for param in op[0].params]
+
+                if not parameters:
```

Comment:
Nothing to do with this PR, but this op-specific logic kind of bothers me.

Reply:
We played around with it, but there was ambiguity because PennyLane and Qiskit treat operations with array input (i.e., `QubitUnitary` and `QubitStateVector`) differently.

Since all plugins currently do this as well (use conditional branches for these two operations), I'm okay with this here --- it doesn't look like there is a standard clean solution to this yet.